app/layout.tsx

```diff
@@ -1,6 +1,10 @@
 import { Links } from "@/components/links";
 import "./globals.css";
 
+// import this to trigger `registerApolloClient` before the components render
+// there might be a better way of doing this, I'm not very familiar with next
+import "@/lib/client";
```

Comment:
This should not be necessary in multiple places - I'm sure next has a central place to register something like this, so this doesn't have to be repeated over the place - I just don't know it yet :)